src/hooks/useUniswapCurrencyList.ts

```diff
@@ -0,0 +1,248 @@
+  const dispatch = useDispatch();
+  const [loading, setLoading] = useState(true);
+  const [curatedAssets, setCuratedAssets] = useState<RT[]>([]);
+  const [favoriteAssets, setFavoriteAssets] = useState<RT[]>([]);
+  const [highLiquidityAssets, setHighLiquidityAssets] = useState<RT[]>([]);
+  const [lowLiquidityAssets, setLowLiquidityAssets] = useState<RT[]>([]);
+  const [unfilteredFavorites, setUnfilteredFavorites] = useState<RT[]>([]);
+  const [verifiedAssets, setVerifiedAssets] = useState<RT[]>([]);
+
+  const curatedMap = useSelector(uniswapCuratedTokensSelector);
+  const favoriteAddresses = useSelector(uniswapFavoritesSelector);
+
+  const handleFavoritesResponse = (favorites: RT[]) => {
+    setUnfilteredFavorites(
+      favorites.map(favorite => {
+        const { address } = favorite;
+        if (address === WETH_ADDRESS) {
+          return {
+            ...favorite,
+            address: ETH_ADDRESS,
+            name: 'Ethereum',
+            symbol: 'ETH',
+            uniqueId: ETH_ADDRESS,
+          };
+        }
+        return favorite;
+      })
+    );
+  };
+
+  const handleVerifiedResponse = useCallback(
+    (tokens: RT[]) => {
+      const addresses = favoriteAddresses.map(a => a.toLowerCase());
+      return tokens.filter(({ address }) => !addresses.includes(address));
+    },
+    [favoriteAddresses]
+  );
+
+  useQuery(
+    ['tokens/uniswap/v2', favoriteAddresses],
+    () =>
+      getUniswapV2Tokens(
+        favoriteAddresses.map(address => {
+          return address === ETH_ADDRESS ? WETH_ADDRESS : address.toLowerCase();
+        })
+      ),
+    {
+      onSuccess: res => handleFavoritesResponse(res),
+    }
+  );
+
+  const getCurated = useCallback(() => {
+    const addresses = favoriteAddresses.map(a => a.toLowerCase());
+    return Object.values(curatedMap).filter(
+      ({ address }) => !addresses.includes(address.toLowerCase())
+    );
+  }, [curatedMap, favoriteAddresses]);
+
+  const getFavorites = useCallback(async () => {
+    return searching
+      ? await searchCurrencyList(unfilteredFavorites, searchQuery)
+      : unfilteredFavorites;
+  }, [searchQuery, searching, unfilteredFavorites]);
+
+  const getResultsForAssetType = useCallback(
+    async (assetType: UniswapCurrencyListType) => {
+      switch (assetType) {
+        case 'verifiedAssets':
+          setVerifiedAssets(
+            handleVerifiedResponse(
+              await searchCurrencyList(assetType, searchQuery)
+            )
+          );
+          break;
+        case 'highLiquidityAssets':
+          setHighLiquidityAssets(
+            await searchCurrencyList(assetType, searchQuery)
+          );
+          break;
+        case 'lowLiquidityAssets':
+          setLowLiquidityAssets(
+            await searchCurrencyList(assetType, searchQuery)
+          );
+          break;
+        case 'favoriteAssets':
+          setFavoriteAssets((await getFavorites()) || []);
+          break;
+        case 'curatedAssets':
```

Comment:
it would be nice to have these as constants somewhere

Reply:
I did not add constants here because `UniswapCurrencyListType` provides code completion and will complain if you use a string outside that union. I'm open to adding more constants if you don't think that's suitable.